global_test.go

```diff
@@ -59,6 +63,34 @@ func TestReplaceGlobals(t *testing.T) {
 	assert.Equal(t, initialS, *S, "Expected func returned from ReplaceGlobals to restore initial S.")
 }
 
+func TestGlobalsConcurrentUse(t *testing.T) {
+	var (
+		stop atomic.Bool
+		wg   sync.WaitGroup
+	)
+
+	for i := 0; i < 100; i++ {
+		wg.Add(2)
+		go func() {
+			for stop.Load() {
```

Comment:
`stop` is going to `false` by default,  so we're never running this loop, or the loop below. condition needs to be flipped.